### PR TITLE
chore: added modal component spec

### PR DESCRIPTION
**Pull request by cdoremus**
Added modal.component.spec.ts with 5 unit tests for modal.component.ts.

### CODE REVIEW

**Comment by cdoremus** [comment]
I'm not sure why these tests pass in circleci without the changes I did in #148. I noticed that when the Angular 2.1 upgrade was done (commit 398a23b013d6a7ba7ec5bd603bdfc8fc20eb6bab) that the tslint version was upgraded from 3.7.1 to 4.0.0-dev.0. It was between these two versions that the changes occurred that I noted in issue #146. Perhaps circleci is caching the old tslint version. Still, I do not know enough about circleci to confirm my suspicion.

**Comment by bodnarbm** [comment]
@cdoremus It looks like the changes in tslint that you are fixing in #148 did not appear in a tagged version of tslint until 3 days ago with 4.0.0-dev.1. Because package.json only requires 4.0.0-dev.0 and CircleCI is caching the node_modules directory, the CircleCI jobs are using tslint 4.0.0-dev.0 (which does work without the changes in #148). However, newer installs of this repo (or installs after node_modules is wiped clean) will install the latest tag for tsline (4.0.0-dev.1), which causes linter for code gov to break as noted.

I am going to open a new issue to track the problem with CircleCI that you noted.

File: src/app/components/modal/modal.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    // call showModal to push an item into the modalActivated$ Observable
+    this.modalService.showModal({description: undefined, title: undefined, url: newUrl});
+    this.fixture.detectChanges();
+
+    let element = this.fixture.debugElement.query(By.css('a'));
+
+    expect(element.nativeElement.href).toBe(newUrl);
+  });
+
+  it('should no longer be visible', () => {
```
Comment (bodnarbm) [inline src/app/components/modal/modal.component.spec.ts]: Same as above with including condition on when the model should no longer be visible.

**Comment by cdoremus** [comment]
I updated the 'should' text with as concise an explanation as I could muster, and expanded this documentation in a comment inside the it() method to avoid the 'should' text from becoming overwhelming wordy. 
 
I should note that the Angular 2 test API can be a steep learning curve and that there is little documentation and example code outside of angular.io. Still, it is a powerful API, and one that I am starting to master. I will continue to contribute test code to this project as time allows.

**Comment by bodnarbm** [comment]
Sorry @cdoremus, I was unclear in my prior comments. I just meant a few more words, like changing `should have a title` to  something like `should display title in h1 when title is set` or changing `should no longer be visible` to `should no longer be visible after model is closed`

I thought your naming was great otherwise.

**Comment by cdoremus** [comment]
I've updated the it() text to make it more concise. I hope this is what you were looking for.

**Pull request merged**